File: Search/WoodCut/WoodCut.cc
class Solution{
    public int woodCut(vector<int> &L, int k) {
        int start = 0, end = 0; 

    }

    private int cut(vector<int> &L, int k, int mid) {
        int sum = 0;
        for(int l : L)
            sum += l / mid;

        return sum >= k;
    }
};

File: Search/WoodCut/WoodCut.py
class Solution:
    def woodCut(self, L, k):
        if sum(L) < K:
            return 0

        start, end = 0, max(L)
        while start + 1 < end:
            mid = start + (end -start) / 2
            pieces = sum(i / mid for i in L)
            if pieces < K:
                end = mid
            else:
                start = mid

        if sum(i / end for i in L) >=k:
            return end
        return start

